ocaml/typing/typedecl.ml

```diff
@@ -63,8 +63,8 @@ type error =
   | Multiple_native_repr_attributes
   | Cannot_unbox_or_untag_type of native_repr_kind
   | Deep_unbox_or_untag_attribute of native_repr_kind
-  | Layout_coherence_check of type_expr * Layout.Violation.t
-  | Layout_update_check of Path.t * Layout.Violation.t
+  | Layout_of_type of type_expr * Layout.Violation.t
+  | Layout_of_path of Path.t * Layout.Violation.t
```

Comment:
These names seem less obviously like errors than the old names - in fact I'm not sure what they mean.  The old names weren't the best either, but maybe we can come up with something that describes the error more clearly?

Reply:
I've added `mismatch`. I don't think it's much worth trying to do better here.